gm4_sweethearts/data/gm4_sweethearts/functions/main.mcfunction

```diff
@@ -0,0 +1,5 @@
+execute as @a[gamemode=!creative,gamemode=!spectator,scores={gm4_sh_sneak=1..},predicate=gm4_sweethearts:holding_flower] at @s run function gm4_sweethearts:sneaking
+scoreboard players reset @a gm4_sh_sneak
```

Comment:
You can also detect sneaking in the predicate, so you probably don't need this scoreboard objective.